bindings/pyroot/ROOT.py

```diff
@@ -383,6 +385,85 @@ def _TTreeAsMatrix(self, columns=None, exclude=None, dtype="double", return_labe
 if (sys.maxsize > 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
     _root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
 
+# RDataFrame.AsNumpy feature
+def _RDataFrameAsNumpy(df, columns=None, exclude=None):
+    """Read-out the RDataFrame as a collection of numpy arrays.
+
+    The values of the dataframe are read out as numpy array of the respective type
+    if the type is a fundamental type such as float or int. If the type of the column
+    is a complex type, such as your custom class or a std::array, the returned numpy
+    array contains Python objects of this type interpreted via PyROOT.
+
+    Be aware that reading out custom types is much less performant than reading out
+    fundamental types such as supported by directly by numpy.
+
+    Note that the reading is performed in multiple threads if the implicit
+    multi-threading of ROOT is enabled.
+
+    Parameters:
+        columns: If None return all branches as columns, otherwise specify names in iterable.
+        exclude: Exclude branches from selection.
+
+    Returns:
+        dict: Dict with column names as keys and 1D numpy arrays with content as values
```

Comment:
It's passed down to the pythonization. So you can see it with `help(...)` or `? ...` or whatever. But ofc our documentation of pythonizations is lacking (in terms of doxygen).

Reply:
I see. In any case could you explicitly mention that calling `AsNumpy` triggers an event loop?

Reply:
Unless we go for saving as private data member the RResultPtrs and we make the full structure lazy as mentioned at the PPP meeting...

Reply:
@etejedor and I have checked. Either we make it lazy or we avoid the extra copy. Soon, I'll make a proposal without extra copy since this speeds up reading custom types by a factor of 4 (only doing the wrapping, not the copy of the complex type saves a lot of time).